### PR TITLE
Implemented N-Queen in Python

### PR DESCRIPTION
**Pull request by subhayu99**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Implemented the N-Queen Problem in Python, added it to the dp folder of Python directory and updated the README.md of Python directory.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Corresponding issue can be found [here](https://github.com/TesseractCoding/NeoAlgo/issues/424)

### CODE REVIEW

File: Python/dp/N-Queen.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
+class QueenChessBoard:
```
Comment (atarax665) [inline Python/dp/N-Queen.py:4]: Add description of the problem at the top.

File: Python/dp/N-Queen.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        self.columns = []
+        self.count = 0
+ 
+    def place_in_next_row(self, column):
+        self.columns.append(column)
+ 
+    def remove_in_current_row(self):
+        return self.columns.pop()
+ 
+    def is_this_column_safe_in_next_row(self, column):
```
Comment (atarax665) [inline Python/dp/N-Queen.py:18]: Use docstrigs to explain what each function does.

File: Python/dp/N-Queen.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            try:
+                prev_column = board.remove_in_current_row()
+            except IndexError:
+                break
+            row -= 1
+            column = 1 + prev_column
+    print('Number of solutions:', number_of_solutions)
+
+n = int(input('Enter no. of Queen: '))
+solve_queen(n)
```
Comment (atarax665) [inline Python/dp/N-Queen.py]: Add sample Input/Output.

**Review by atarax665 — CHANGES_REQUESTED** [review]

File: Python/dp/N-Queen.py (hunk 1 above)
Comment (subhayu99) [reply Python/dp/N-Queen.py:4]: Done

File: Python/dp/N-Queen.py (hunk 2 above)
Comment (subhayu99) [reply Python/dp/N-Queen.py:18]: Done

File: Python/dp/N-Queen.py (hunk 3 above)
Comment (subhayu99) [reply Python/dp/N-Queen.py]: Done

**Comment by subhayu99** [comment]
@atarax665 Can you please tell me why is this Merge Conflicts occuring?

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@subhayu99 Do not edit `C/README.md`. Format the code according to PEP 8 rules. Refer this http://pep8online.com/.

**Comment by subhayu99** [comment]
@iamrajiv I have edited the `C/README.md` because it was showing error.
Also, I would like to ask you about which code are asking me to format as per the PEP 8 rules.

**Comment by iamrajiv** [comment]
@subhayu99 Format `N-Queen.py` code. Also, do not edit anything in `C/README.md` file it is not related to this program.

**Comment by subhayu99** [comment]
Yeah I know that `C/README.md` isn't related but it showed me some error

**Comment by subhayu99** [comment]
@iamrajiv Okay, then can you close this pull request... I'll put up another pull request then

**Comment by iamrajiv** [comment]
@subhayu99 What error can you elaborate?

**Comment by subhayu99** [comment]
There was some typos in the file and that was causing the merge error. I don't think I have removed something valuable 😅

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@subhayu99 Still `C/README.md` the file is there do not add this file in this PR because `C/README.md` is formatted recently and this update will cause a problem so better exclude this from PR.

**Comment by subhayu99** [comment]
@iamrajiv Is there a way to remove the `C/README.md` from this PR? If so, please let me know, as I don't have any idea of that.

**Comment by iamrajiv** [comment]
@subhayu99 I have even the `C/README.md` file with `C/README.md` of `TesseractCoding:master` branch so there will be no file changed in `C/README.md`.

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@subhayu99 Format `N-Queen.py` program according to PEP 8 rules.

**Comment by subhayu99** [comment]
@iamrajiv I have formatted the `N-Queen.py` as per the PEP 8 style... But I don't know what went wrong with the `C-Plus-Plus/sort/Shellsort.cpp` file. If you can then please help me.

**Review by subhayu99 — COMMENTED** [review]
.

**Review by iamrajiv — APPROVED** [review]

**Comment by iamrajiv** [comment]
@subhayu99 I resolve the issue. Also, I would suggest making a separate branch for every new PR and avoid using `master` branch.

**Comment by iamrajiv** [comment]
@subhayu99 I have resolved the issue. Also, I would suggest making a separate branch for every new PR and avoid using `master` branch.

**Comment by subhayu99** [comment]
@iamrajiv Thanks a lot for the help and I will surely follow the steps you said.

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
@subhayu99 can you please tell me why am I seeing so many change sin Readme file? Also please add a sample input/output and space and time complexity in commented form below the code section.

**Comment by subhayu99** [comment]
@Kajol-Kumari To be honest, I don't know why all these changes occured in the Readme file. I have contributed to this project earlier also, but that time I wasn't asked for so many changes repetitively and also it was merged very early, within 3-5 days to be specific. Also, I have seen some other scripts in the repo and I haven't seen space and time complexities added to the code. I'm saying all these just because it's been 21 days now and everytime I did some changes either an error occurred or else I'm asked to do some unnecessary additions to the code and the cycle goes on. 😑

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@subhayu99 Just copy the whole content of https://github.com/TesseractCoding/NeoAlgo/blob/master/Python/README.md and paste it to your https://github.com/subhayu99/NeoAlgo/blob/master/Python/README.md. In addition to this add your program link in readme this will work fine.

**Pull request closed without merging**